src/Integration.Vsix/Analysis/IssueConsumerFactory_IssueHandler.cs

```diff
@@ -0,0 +1,116 @@
+
+            private readonly ITextDocument textDocument;
+            private readonly string projectName;
+            private readonly Guid projectGuid;
+            private readonly IIssuesFilter issuesFilter;
+            private readonly PublishSnapshot publishSnapshot;
+
+            private readonly TranslateSpans translateSpans;
+
+            public IssueHandler(ITextDocument textDocument,
+                string projectName,
+                Guid projectGuid,
+                IIssuesFilter issuesFilter,
+                PublishSnapshot publishSnapshot)
+                : this (textDocument, projectName, projectGuid, issuesFilter, publishSnapshot, DoTranslateSpans)
+            {
+            }
+
+            internal /* for testing */ IssueHandler(ITextDocument textDocument,
+                string projectName,
+                Guid projectGuid,
+                IIssuesFilter issuesFilter,
+                PublishSnapshot publishSnapshot,
+                TranslateSpans translateSpans)
+            {
+                this.textDocument = textDocument;
+                this.projectName = projectName;
+                this.projectGuid = projectGuid;
+                this.issuesFilter = issuesFilter;
+                this.publishSnapshot = publishSnapshot;
+
+                this.translateSpans = translateSpans;
+            }
+
+            internal /* for testing */ void HandleNewIssues(IEnumerable<IAnalysisIssueVisualization> issues)
+            {
+                var filteredIssues = RemoveSuppressedIssues(issues);
+
+                // The text buffer might have changed since the analysis was triggered, so translate
+                // all issues to the current snapshot.
+                // See bug #1487: https://github.com/SonarSource/sonarlint-visualstudio/issues/1487
+                var translatedIssues = translateSpans(filteredIssues, textDocument.TextBuffer.CurrentSnapshot);
+
+                var newSnapshot = new IssuesSnapshot(projectName, projectGuid, textDocument.FilePath, translatedIssues);
+                publishSnapshot(newSnapshot);
+            }
+
+            private IEnumerable<IAnalysisIssueVisualization> RemoveSuppressedIssues(IEnumerable<IAnalysisIssueVisualization> issues)
+            {
+                var filterableIssues = issues.OfType<IFilterableIssue>().ToArray();
+
+                var filteredIssues = issuesFilter.Filter(filterableIssues);
+                Debug.Assert(filteredIssues.All(x => x is IAnalysisIssueVisualization), "Not expecting the issue filter to change the list item type");
+
+                return filteredIssues.OfType<IAnalysisIssueVisualization>().ToArray();
+            }
+
+            private static IEnumerable<IAnalysisIssueVisualization> DoTranslateSpans(IEnumerable<IAnalysisIssueVisualization> issues, ITextSnapshot activeSnapshot)
```

Comment:
Renamed from `TranslateSpans` to `DoTranslateSpans` - avoids name collision with the new `TranslateSpans` delegate.